maxBufferSize redeclared / overflows int

Follows https://github.com/ssbc/go-ssb-room/pull/355#issuecomment-1312053857, https://github.com/ssbc/go-muxrpc/pull/16 was not a good fix after all :woman_facepalming: 

```
GOARCH=arm GOOS=linux go build ./...
# github.com/ssbc/go-muxrpc/v2/codec
codec/buffer_32bit.go:10:7: maxBufferSize redeclared in this block
	codec/buffer.go:14:7: other declaration of maxBufferSize
codec/writer.go:26:15: maxBufferSize (untyped int constant 4294967295) overflows int
```

```
go vet ./codec
# github.com/ssbc/go-muxrpc/v2/codec
vet: codec/buffer_32bit.go:10:7: maxBufferSize redeclared in this block
```

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -29,8 +29,11 @@ jobs:
         cache: 'npm'
         cache-dependency-path: '**/package-lock.json'
 
-    - name: Build
+    - name: Build smoke test
       run: go build -v ./...
+
+    - name: Build arm smoke test
+      run: GOARCH=arm go build -v ./...
 
     - name: Install node ssb-stack
       run: npm ci --prefer-offline --no-audit

--- a/codec/buffer.go
+++ b/codec/buffer.go
@@ -2,8 +2,8 @@
 //
 // SPDX-License-Identifier: MIT
 
-//go:build !386 || !arm
-// +build !386 !arm
+//go:build !(386 || arm)
+// +build !386,!arm
 
 package codec
 

